### PR TITLE
ship linting

### PR DESCRIPTION
**Pull request by dexhorthy**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "replicated-lint",
   "author": "Replicated, Inc.",
-  "version": "0.10.8",
```
Comment (laverya) [inline package.json:4]: Want to update the changelog too?

File: src/engine.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import * as _ from "lodash";
 import { Predicate, Test } from "./lint";
 import {
-  AdminCommandContainerMissing,
+  AdminCommandContainerMissing, AllOf,
```
Comment (laverya) [inline src/engine.ts]: I feel like this should be two lines

File: src/engine.ts [hunk 3]
```diff
@@ -59,6 +59,7 @@ const defaultPredicates: PredicateRegistry = {
   AdminCommandContainerMissing,
   And,
   AnyOf,
+  AllOf,
```
Comment (laverya) [inline src/engine.ts]: And that this should be before `And`
Not that we seem to be caring about alphabetizing things anymore, but still

File: projects/replicated-ship/rules.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+- name: spec-require-render
+  type: error
+  message: lifecycle should include at least one `render` step
+  test:
+    AllOf:
```
Comment (laverya) [inline projects/replicated-ship/rules.yaml:6]: I don't think I understand `AllOf` here
Do you want all lifecycles to have a render step?
Or at least one to have that?

If you're testing multiple lifecycles, you should make test cases with multiple lifecycles
If there can only be one, then we don't need `AllOf` - Just `Exists` iirc

**Review by laverya — CHANGES_REQUESTED** [review]
We need to clarify that `AllOf` rule
And also add to the changelog

File: projects/replicated-ship/rules.yaml (hunk 4 above)
Comment (dexhorthy) [reply projects/replicated-ship/rules.yaml:6]: How would you write a rule that "at least one of the lifecycle steps has a truthy `render` field" using Exists?

Comment (laverya) [reply projects/replicated-ship/rules.yaml:6]: This looks much better

**Review by laverya — APPROVED** [review]

**Pull request merged**